src/ast/passes/codegen_llvm.cpp

```diff
@@ -1026,6 +1026,15 @@ void CodegenLLVM::visit(Call &call)
   {
     expr_ = b_.getInt64(call.vargs->at(0)->type.GetSize());
   }
+  else if (call.func == "offsetof")
+  {
+    auto &element_arg = *call.vargs->at(1);
+    String &element = static_cast<String&>(element_arg);
+    if (!call.vargs->at(0)->type.HasField(element.str))
+      LOG(ERROR) << "struct has no field named '" + element.str + "'";
```

Comment:
Could this check be in semantic analyser? It'd make more sense there and you can add location info to the error.